packages/user-operation-controller/src/UserOperationController.ts

```diff
@@ -517,4 +660,143 @@ export class UserOperationController extends BaseController<
       },
     );
   }
+
+  async #requestApproval(metadata: UserOperationMetadata): Promise<AddResult> {
+    const { id, origin } = metadata;
+    const type = ApprovalType.Transaction;
+    const requestData = { txId: id };
+
+    return (await this.messagingSystem.call(
+      'ApprovalController:addRequest',
+      {
+        id,
+        origin,
+        type,
+        requestData,
+        expectsResult: true,
+      },
+      true, // Should display approval request to user
+    )) as Promise<AddResult>;
+  }
+
+  async #getTransactionType(
+    transaction: TransactionParams | undefined,
+    provider: Provider,
+  ): Promise<TransactionType | undefined> {
+    if (!transaction) {
+      return undefined;
+    }
+
+    const ethQuery = new EthQuery(provider);
+    const result = determineTransactionType(transaction, ethQuery);
+
+    return (await result).type;
+  }
+
+  async #getProvider(
+    networkClientId: string,
+  ): Promise<{ provider: Provider; chainId: string }> {
+    const { provider, configuration } = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      networkClientId,
+    );
+
+    const { chainId } = configuration;
+
+    return { provider, chainId };
+  }
+
+  async #updateUserOperationAfterApproval(
+    metadata: UserOperationMetadata,
+    request: AddUserOperationRequest,
+    options: AddUserOperationOptions,
+    updatedTransaction: TransactionMeta,
+  ) {
+    log('Found updated transaction in approval', { updatedTransaction });
+
+    const { userOperation } = metadata;
+    const usingPaymaster = userOperation.paymasterAndData !== EMPTY_BYTES;
+
+    const updatedMaxFeePerGas = addHexPrefix(
+      updatedTransaction.txParams.maxFeePerGas as string,
+    );
+
+    const updatedMaxPriorityFeePerGas = addHexPrefix(
+      updatedTransaction.txParams.maxPriorityFeePerGas as string,
+    );
+
+    let refreshUserOperation = false;
+    const previousMaxFeePerGas = userOperation.maxFeePerGas;
```

Comment:
Matt, I noticed we're using three different terms in this method that essentially mean the same thing: "update", "refresh", and "regenerate" for UserOperation. It might be clearer if we stick to just one.

How about we go with "update"? We could then rename the method to "checkAndUpdateUserOpAfterApproval", call the variable "updateNeeded", and change "#regenerateUserOperation" to "#updateUserOperation". This should make our code more consistent and easier to follow.

What do you think?

Reply:
This was intentional as we use `update` generally across controllers to refer to state update, but I needed a term to summarise the process of preparing the user operation again, updating gas and gas fees, and adding paymaster data.

`Regenerate` seemed logical since we're generating our user operation for the second time.

`refreshUserOperation` was accidental though so I've renamed that to `regenerateUserOperation`.